base/compiler/ssair/passes.jl

```diff
@@ -937,10 +939,11 @@ function pattern_match_typeof(compact, typ, fidx, Targ, farg)
         push!(tvars, applyTvar)
     end
 
-    @assert applyT.name === typ.name
+    applyT.name === typ.name || return false
```

Comment:
Did you encounter a situation where this happens? `typ` is supposed to be the immediate result of the `Targ` computation, so I didn't think this was possible.

Reply:
Yeah on a second look I think this check should probably be assertion instead.